Engine/script/cc_instance.cpp

```diff
@@ -80,38 +80,38 @@ struct ScriptCommandInfo
+    ScriptCommandInfo( SCMD_MEMWRITE        , "memwrite4"         , 1, kScOpOneArgIsReg ),
     ScriptCommandInfo( SCMD_MULREG          , "mul"               , 2, kScOpTwoArgsAreReg ),
     ScriptCommandInfo( SCMD_DIVREG          , "div"               , 2, kScOpTwoArgsAreReg ),
     ScriptCommandInfo( SCMD_ADDREG          , "add"               , 2, kScOpTwoArgsAreReg ),
     ScriptCommandInfo( SCMD_SUBREG          , "sub"               , 2, kScOpTwoArgsAreReg ),
-    ScriptCommandInfo( SCMD_BITAND          , "bit_and"           , 2, kScOpTwoArgsAreReg ),
-    ScriptCommandInfo( SCMD_BITOR           , "bit_or"            , 2, kScOpTwoArgsAreReg ),
-    ScriptCommandInfo( SCMD_ISEQUAL         , "cmp"               , 2, kScOpTwoArgsAreReg ),
-    ScriptCommandInfo( SCMD_NOTEQUAL        , "ncmp"              , 2, kScOpTwoArgsAreReg ),
+    ScriptCommandInfo( SCMD_BITAND          , "and"               , 2, kScOpTwoArgsAreReg ),
+    ScriptCommandInfo( SCMD_BITOR           , "or"                , 2, kScOpTwoArgsAreReg ),
+    ScriptCommandInfo( SCMD_ISEQUAL         , "cmpeq"             , 2, kScOpTwoArgsAreReg ),
+    ScriptCommandInfo( SCMD_NOTEQUAL        , "cmpne"             , 2, kScOpTwoArgsAreReg ),
     ScriptCommandInfo( SCMD_GREATER         , "gt"                , 2, kScOpTwoArgsAreReg ),
     ScriptCommandInfo( SCMD_LESSTHAN        , "lt"                , 2, kScOpTwoArgsAreReg ),
     ScriptCommandInfo( SCMD_GTE             , "gte"               , 2, kScOpTwoArgsAreReg ),
     ScriptCommandInfo( SCMD_LTE             , "lte"               , 2, kScOpTwoArgsAreReg ),
-    ScriptCommandInfo( SCMD_AND             , "and"               , 2, kScOpTwoArgsAreReg ),
-    ScriptCommandInfo( SCMD_OR              , "or"                , 2, kScOpTwoArgsAreReg ),
+    ScriptCommandInfo( SCMD_AND             , "land"              , 2, kScOpTwoArgsAreReg ),
+    ScriptCommandInfo( SCMD_OR              , "lor"               , 2, kScOpTwoArgsAreReg ),
     ScriptCommandInfo( SCMD_CALL            , "call"              , 1, kScOpOneArgIsReg ),
-    ScriptCommandInfo( SCMD_MEMREADB        , "memread.b"         , 1, kScOpOneArgIsReg ),
-    ScriptCommandInfo( SCMD_MEMREADW        , "memread.w"         , 1, kScOpOneArgIsReg ),
-    ScriptCommandInfo( SCMD_MEMWRITEB       , "memwrite.b"        , 1, kScOpOneArgIsReg ),
-    ScriptCommandInfo( SCMD_MEMWRITEW       , "memwrite.w"        , 1, kScOpOneArgIsReg ),
-    ScriptCommandInfo( SCMD_JZ              , "jz"                , 1, kScOpNoArgIsReg ),
+    ScriptCommandInfo( SCMD_MEMREADB        , "memread1"          , 1, kScOpOneArgIsReg ),
+    ScriptCommandInfo( SCMD_MEMREADW        , "memread2"          , 1, kScOpOneArgIsReg ),
+    ScriptCommandInfo( SCMD_MEMWRITEB       , "memwrite1"         , 1, kScOpOneArgIsReg ),
+    ScriptCommandInfo( SCMD_MEMWRITEW       , "memwrite2"         , 1, kScOpOneArgIsReg ),
+    ScriptCommandInfo( SCMD_JZ              , "jzi"               , 1, kScOpNoArgIsReg ),
```

Comment:
This should stay `jz`, it doesn't really accept a literal but an offset relative to the instruction pointer.

Reply:
no, it takes an absolute code address in the current script

Reply:
Are you sure? Then it must have changed _quite_ recently. I'm working with release 3.5.2, and here's the code for the JZ command:
```
case SCMD_JZ:
          if (registers[SREG_AX].IsNull())
              pc += arg1.IValue;
          break;
```

Reply:
oh ? maybe i'm mistaken here, or maybe the engine changes the relocation from absolute to relative on fixup; can't recall.
it doesn't really matter anyhow, the value is an immediate whether absolute or relative, as opposed to a register number.

Reply:
> maybe i'm mistaken here, or maybe the engine changes the relocation from absolute to relative on fixup; can't recall.

It's stored as relative in the byte code as well. The problem I see here is if we decide to add indirect jumps, we have to define more jmp mnemonic variations. The reader should be able to determine what type of jump it is from the arguments.

Reply:
> problem I see here is if we decide to add indirect jumps, we have to define more jmp mnemonic variations

yes of course. if you add new instructions, you logically also have to add a new mnemonic - at least if you want to have one mnemonic per instruction, which eases readibility and codegen.

> The reader should be able to determine what type of jump it is from the arguments.

well first of all, there aren't really that many readers of asm (i suspect @fernewelten and i are so far the only ones that spent more than an hour reading asm). second, if the mnemonic adds a hint about what the instruction does, it certainly helps readibility, even if theoretically one can guess from the argument what happens.
thirdly, i can't see a good reason why one would like to add more jump types. i can imagine a lot of instructions that (when used by the compiler) could make scripts a good bit faster, but none of them involve jumps.
for example, if there was an instruction `cmpeqi reg, value`, the usual tuple of `li ax, value; cmpeq ax, bx`, often accompanied by register swaps or push/pops could be replaced with a single operation. i have a number of opcodes in mind that could give a good speed boost, but that is a matter for another PR/discussion.

Reply:
As far as I can see, the compiler has generated loops that needed to be relocated afterwards, so that was probably the reason to decide that relative jumps are easier to work with. 

Proper assemblers used to have jump labels and jumps to labels, and if the jump instruction was relative, it was the assembler's job to figure out the relative distance. At least that's what I recollect from the z80 / TRS-80 era, ages ago.